solar-system.rb

```diff
@@ -0,0 +1,310 @@
+#     :rule => "first order"}
+# },
+# { :Naboo => {
+#     :color => "gree",
+#     :temp => 281,
+#     :size => "big",
+#     :diameter => 14563,
+#     :rule => "resistance"}
+# }])
+#
+# puts angelas_solar_system.summary
+
+######### WAVE 2 ############
+
+# # Create a Planet class which will represent a planet.
+# # Add an initialize method which takes several arguments and uses them to set the class' instance variables.
+# # Create a method that returns the Planet's attributes in an easy to read fashion.
+# # Create reader methods to give a user access to read the instance variables.
+# # Make your SolarSystem class take an array of Planets instead of hashes.
+# # When printing the planet list or planet details, it should call the corresponding method in Planet.
+#
+#
+# class SolarSystem
+#
+#   attr_accessor :planets
+#
+#   def initialize(input_planets)
+#     @planets = input_planets
+#
+#     # ## optional
+#     # one_instance_of_a_planet = @planets[0]
+#     # one_instance_of_a_planet.print_out
+#   end
+#
+#   def summary
+#     planet_info = []
+#     @planets.each do |planet|
+#         planet_info << ("#{planet.print_out}")
+#       end
+#     return planet_info
+#   end
+#
+# end
+#
+# class Planet
+#
+#   attr_reader   :name, :distance_from_core, :native_species, :moons, :primary_export
+#
+#   def initialize(name, distance_from_core, native_species, moons, primary_export)
+#     @name = name
+#     @distance_from_core = distance_from_core #in light years
+#     @native_species = native_species
+#     @moons = moons
+#     @primary_export = primary_export
+#   end
+#
+#   #method prints out all info about given planet:
+#   def print_out
+#     return "✳︎ #{@name} ✳︎\n" +
+#    " ⭐︎ #{@name} is #{@distance_from_core} light years away from core.\n" +
+#    " ⭐︎ The native species of #{@name} are #{@native_species}.\n" +
+#    " ⭐︎ #{@name} has #{@moons} moons.\n" +
+#    " ⭐︎ #{@name}'s primary export: #{@primary_export}.\n\n"
+#     end
+#
+# end
+#
+# #array of planets that will go through Planet class
+# #name, distance from core, native species, # of moons, primary export
+#
+# planet_array = [
+#   Planet.new("Endor", 16000, "Ewoks", 0, "Medicinals"),
+#   Planet.new("Alderaan", 5000, "Killiks", 1, "Wine & Art"),
+#   Planet.new("Coruscant", 10000, "Humans/Zhell", 4,"None - governing body"),
+#   Planet.new("Jakku", 240000, "Teedus", 1, "Bezorite & junk metal"),
+#   Planet.new("Hoth", 50200, "Skels", 3, "Ice")
+# ]
+#
+#
+# # planet_array.each do |planet_info|
+# #   puts planet_info.print_out
+# # end
+#
+# #long way:
+# # puts planet_array[0].print_out
+# # puts planet_array[1].print_out
+# # puts planet_array[2].print_out
+# # puts planet_array[3].print_out
+# # puts planet_array[4].print_out
+#
+#
+# star_wars_solar_system = SolarSystem.new(planet_array)
+#
+# puts star_wars_solar_system.summary
+
+################### WAVE 3 ########################
+
+
+class SolarSystem
+
+  attr_accessor :planets
```

Comment:
Does this need to be an accessor?  Would it not be better to make it a reader, or not provide an accessor at all?